### DISCUSSION TITLE
How to use included DNS/unbound?

### DISCUSSION THREAD

**Discussion by DennisGaida** [qdm12/gluetun#921] [Q&A] [answered]
I'm unsure how to make use of the included DNS / unbound feature and couldn't find it documented anywhere besides setting a bunch of DNS options: https://github.com/qdm12/gluetun/wiki/DNS-options

Is this DNS supposed to be internal to the container, i.e. used by the VPN? Should I use the DNS from other containers - if so, wouldn't I need to set the DNS server explicitly in the docker-compose of these other containers since otherwise it would default to `127.0.0.1#53`? Shouldn't I also expose the DNS port of the gluetun container?

A lot of basic questions, but I am really unsure what to do with the DNS, though it sounds interesting.

**Comment by qdm12** [comment, marked as answer]
You can just leave it as default. The unbound server runs and listens on port 53, so the default uses DNS_ADDRESS=127.0.0.1:53 in order to use that server.

You could publish port 53 from gluetun to have your home/local DNS go through the vpn + dns over tls if you want.

You may want to subscribe to #137 which will use https://github.com/qdm12/dns/tree/v2.0.0-beta which I have coded from scratch. I'm currently working on it now so it should be done soon. It offers a few things to on top of Unbound, notably less memory usage when blocking some lists.

**Reply by DennisGaida** [reply]
Really interesting, so right now it is supposed to be internal to the container and to be used by the VPN - gotcha!

On https://github.com/qdm12/dns/tree/v2.0.0-beta - looks really clean, but I (as well other many others) am using Pihole for my DNS needs which already includes an unbound instance. I don't know why I would use yet another container. Since you are going into a similar direction with hostnames/IP blocking, you should at least guide users on why to use your image compared to Pihole or Adguard Home (see https://www.smarthomebeginner.com/pi-hole-vs-adguard-home/ for a good comparison)

**Reply by qdm12** [reply]
To be fair Adguard home is quite more awesome than pihole or my own program.

The main difference I would say with adguard is my repo is aimed as a Go API which makes it easily pluggable in gluetun to have a single binary. But end-user wise, yeah it doesn't add anything I think. I'll dig into comparing it more once I finish it out of beta 😉
